e2e-tests/smart-update/run

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+set -o errexit
+set -o xtrace
+
+test_dir=$(realpath $(dirname $0))
+. ${test_dir}/../functions
+
+cluster="smart-update"
+create_namespace $namespace
+deploy_operator
+
+kubectl_bin apply -f $conf_dir/secrets.yml -f $conf_dir/client.yml
+
+IMAGE_MONGOD="perconalab/percona-server-mongodb-operator:master-mongod4.0"
+IMAGE_MONGOD_TO_UPDATE="perconalab/percona-server-mongodb-operator:master-mongod4.2"
```

Comment:
Maybe it is bad idea to hardcode it, we need to have possibility to test the update for different images. So by default we can use it if env variables were not set.

Reply:
But we already hardcode a lot of variables (in function file)
In my opinion this test should check how one-failover-update work. And it does not depend on what versions we are using.

Well, actually it can depend on it if we use images that can be updated from one to another, but it does not relate to one-failover-update implementation.

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/390/commits/4882291abae0b4523f85b20da53485ec49783351